code/asteroid/asteroid.cpp

```diff
@@ -733,52 +766,56 @@ static void asteroid_aim_at_target(object *objp, object *asteroid_objp, float de
 }
 
 /**
- * Call once per frame to maybe throw an asteroid at a ship.
+ * Call once per frame to maybe throw some asteroids at one or more ships.
  *
- * @param count asteroids already targeted on
  */
-static void maybe_throw_asteroid(int count)
+static void maybe_throw_asteroid()
 {
-	if (!timestamp_elapsed(Next_asteroid_throw)) {
-		return;
-	}
+	//sanitize the list
+	for (size_t i = 0; i < Asteroid_targets.size();) {
+		object* target_objp = &Objects[Asteroid_targets[i].objnum];
+		if (target_objp->type == OBJ_NONE || target_objp->signature != Asteroid_targets[i].signature) {
+			Asteroid_targets[i] = Asteroid_targets.back();
+			Asteroid_targets.pop_back();
+		} else  // if we needed to cull we should not advance because we just moved a new asteroid target into this spot
+			i++;
+	}
+
+	for (asteroid_target& target : Asteroid_targets) {
+		if (!timestamp_elapsed(target.throw_stamp))
+			return;
```

Comment:
So if the first target's timestamp is not elapsed, no asteroid should be thrown this frame at all?

Reply:
Whoops! should be `continue`

Reply:
![istockphoto-183826186-1024x1024](https://user-images.githubusercontent.com/12529946/142715148-cf215cb4-0438-4051-baa2-51975b131c2b.jpg)